soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -323,6 +332,13 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 	ARG_UNUSED(substate_id);
 	uint32_t cpu = arch_proc_id();
 
+	if (cpu == 0) {
+		int ret = pm_device_runtime_get(INTEL_ADSP_HST_DOMAIN_DEV);
+
+		ARG_UNUSED(ret);
+		__ASSERT(ret == 0, "HST power domain resume operation failed.");
```

Comment:
Nitpick: assertions will be disabled in production code.  If you have a runtime error message like this, it implies that you want to handle this, in which case it should be a k_panic() or similar.  If you're sure this driver can't fail (looks like it's just a bit set, so probably), then I'd avoid having the string in the image (we want assertions to be so lightweight that no one worries about turning them on!) with e.g. `___ASSERT_NO_MSG(ret == 0); /* can't happen with this driver */`

Ditto for the other asserts in this patch.

Reply:
Maybe you're just trying to be nice but I wouldn't call this a "nitpick". Either we want to catch this error in all configurations and it should not be an assert, or it's just a debugging aid and it can be an assert (I don't know the answer here). This decision has both CPU time and (even more) "human time" consequences.

Reply:
@andyross Yes, you're right. I don't expect any error to appear here. Thanks for the suggestions.